### PR TITLE
[feat] Change WYSIWYG editor in run detail notes tab

### PR DESCRIPTION
**Pull request by arsengit**
- Changed `Notes editor` from `Toast UI` to `Rich Markdown Editor`
- Fixed `Error handling` while requesting with the wrong `run id`.
- Added `RouteLeavingGuard` component for preventing route change.
- Updated `react-router-dom` to 5.3.0

### CODE REVIEW

**Comment by arsengit** [comment]
Resolves #291

File: aim/web/ui/src/services/models/notes/notesModel.ts [hunk 1]
```diff
@@ -25,17 +25,14 @@ const model = createModel<Partial<INotesAppModelState>>({
...
-let getNotesListRequestRef: IApiRequestRef<any>;
+let getNotesListRequestRef: IApiRequest<any>;
 
 // Initializing model
 function initialize(runId: string): void {
   model.init();
   try {
     getNotesListRequestRef = onNotesListFetch(runId);
-    getNotesListRequestRef.call((detail: any) => {
-      exceptionHandler({ detail, model });
```
Comment (VkoHov) [inline aim/web/ui/src/services/models/notes/notesModel.ts:36]: Why have you deleted the `exceptionHandler` method here?

Comment (VkoHov) [inline aim/web/ui/src/services/models/notes/notesModel.ts:36]: Why have you deleted the `exceptionHandler` method here?

File: aim/web/ui/src/pages/RunDetail/RunDetailNotesTab/RunDetailNotesTab.tsx [hunk 2]
```diff
@@ -42,88 +43,104 @@ function RunDetailNotesTab({ runHash }: IRunDetailNotesTabProps) {
...
-          initialEditType='markdown'
-          height='calc(100vh - 146px)'
-          ref={editorRef}
-        />
-        <div className='RunDetailNotesTab__Editor__actionBtns'>
-          <Tooltip title='Delete Note'>
+        <div className='RunDetailNotesTab__Editor__actionPanel'>
+          <div className='RunDetailNotesTab__Editor__actionPanel__info'>
+            {noteData?.created_at && (
+              <Tooltip title='Created At'>
```
Comment (VkoHov) [inline aim/web/ui/src/pages/RunDetail/RunDetailNotesTab/RunDetailNotesTab.tsx:94]: Can't we have reuseable component for both `Created At` and `Updated At`

Comment (VkoHov) [inline aim/web/ui/src/pages/RunDetail/RunDetailNotesTab/RunDetailNotesTab.tsx:94]: Can't we have reuseable component for both `Created At` and `Updated At`

**Comment by VkoHov** [comment]
I can't differ the selected state and the hovered state of the action popup.
Screen Shot 2022-04-25 at 17 28 22

**Comment by VkoHov** [comment]
There is an inconsistency with H1, H2, and H3.
H2 is bigger than H1.
H3 is the same size as H1.
Screen Shot 2022-04-25 at 18 15 17

**Comment by VkoHov** [comment]
Let's change `At` to lowercase.
Screen Shot 2022-04-25 at 18 27 17
Screen Shot 2022-04-25 at 18 27 08

And maybe we can change the updated date icon to a more intuitive one? For E.g. see in the screenshot.
Screen Shot 2022-04-25 at 18 30 24

**Comment by VkoHov** [comment]
Is it a bug or we are ok with that?

https://user-images.githubusercontent.com/43929589/165111824-5cde60e0-7429-4023-8493-afa9ce7c5efa.mov

**Review by VkoHov — APPROVED** [review]

**Comment by VkoHov** [comment]
lgtm!

**Review by roubkar — APPROVED** [review]

**Pull request merged**